components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -297,8 +372,13 @@ impl<ER: RaftEngine> Storage<ER> {
         if let Some(hs) = ready.hs() {
             entry_storage.raft_state_mut().set_hard_state(hs.clone());
         }
-        if prev_raft_state != *entry_storage.raft_state() {
+        if !ever_persisted || prev_raft_state != *entry_storage.raft_state() {
             write_task.raft_state = Some(entry_storage.raft_state().clone());
         }
+        if !ever_persisted {
```

Comment:
Question: So ever_persisted is false when it's created by uninit().  Will this peer repeatedly be called with handle_raft_ready? If that the region state will be write multiple times.  
How is the lifetime of the Peer created by uninit()?

Reply:
You are right. `ever_persisted` is supposed to be reset after first write.